marlin-succinct/protocol/src/prover.rs

```diff
@@ -134,29 +134,29 @@ impl<E: PairingEngine> ProverProof<E>
 
         // the transcript of the random oracle non-interactive argument
         // the first part of the argument needs the following oracles:
-        // alpha, betta[0] in order to compute RA and evaluate Z, ZA & ZB polynomials
+        // alpha, beta[0] in order to compute RA and evaluate Z, ZA & ZB polynomials
 
         let mut argument = RandomOracleArgument::<E::Fr>::new(index.oracle_params.clone());
         
         argument.commit_scalar(&E::Fr::one()); // this commit comes from the previous proof context
         oracles.alpha = argument.challenge();
         // commit the public input into the argument
         argument.commit_slice(&witness.0[0..witness.1]);
-        oracles.betta[0] = argument.challenge();
+        oracles.beta[0] = argument.challenge();
 
         // evaluate Z, ZA & ZB polynomials
-        w_eval = w.evaluate(oracles.betta[0]);
-        za_eval = za.evaluate(oracles.betta[0]);
-        zb_eval = zb.evaluate(oracles.betta[0]);
+        w_eval = w.evaluate(oracles.beta[0]);
+        za_eval = za.evaluate(oracles.beta[0]);
+        zb_eval = zb.evaluate(oracles.beta[0]);
 
         // run the random oracle argument further to query
-        // oracles gamma[0..3]
+        // oracles eta[0..3]
         argument.commit_scalar(&w_eval);
-        oracles.gamma[0] = argument.challenge();
+        oracles.eta[0] = argument.challenge();
```

Comment:
You can absorb the commitments to `w, z_A, z_B` before squeezing `alpha` and the `eta` values -- it's better for efficiency to not switch between absorbing and squeezing.

Reply:
Oh actually I don't quite understand what's going on, it seems things are not in the correct order here.

Reply:
I believe the order should be as in https://github.com/CodaProtocol/coda/blob/double-marlin-verifier/src/lib/double_marlin/double_marlin.ml#L202. And it is not necessary to absorb any polynomial evalutions